Modules/socketmodule.c

```diff
@@ -1227,6 +1227,14 @@ makesockaddr(SOCKET_T sockfd, struct sockaddr *addr, size_t addrlen, int proto)
        }
 #endif /* AF_NETLINK */
 
+#if defined(AF_VSOCK)
+       case AF_VSOCK:
+       {
+           struct sockaddr_vm *a = (struct sockaddr_vm *) addr;
+           return Py_BuildValue("II", a->svm_cid, a->svm_port);
```

Comment:
You don't have the proper headers. 
checking for linux/vm_sockets.h... no
 
So it looks like my checking for HAVE_LINUX_VM_SOCKETS_H is faulty.